Add demo project for esp32

kaans opened the pull request:
This PR adds a demo project for the ESP32. It is inspired by the demo project for the ESP8266 and adjusted to the requirements for ESP32. It is tested with the current v4.4 release and an ESP32 with 4 MB flash.

I had to introduce a new define macro indicating that the ESP32 demo is running. The FDB_INFO macros in tsdb_sample.c use an incompatible type for the _tsl->time_ on the ESP32 plattform. If 32 bit time type is used (_FDB_USING_TIMESTAMP_64BIT_ not defined), then the type is _int_. If 64 bit time type is used (_FDB_USING_TIMESTAMP_64BIT_ is defined), then the type is _long long int_. For the demo I assume 32bit time type.

armink commented:
Thx for your contribution. 👍 

About the timestrap print compatibility. May be you can change the print code to:

```c
#ifdef FDB_USING_TIMESTAMP_64BIT 
#define __PRITS "ld"
#else
#define __PRITS "d"
#endif
....
FDB_INFO("[query_cb] queried a TSL: time: %" __PRITS ", temp: %d, humi: %d\n", tsl->time, status.temp, status.humi);
```

kaans commented:
Thanks for your suggestion, I adjusted the code accordingly and removed the esp32 specific macro form the samples. With this change the demo works if 32 bit is used for the time.
